### ISSUE TITLE
anchor in timeline item does not work in click (Ipad)

### ISSUE THREAD

**Issue by schotman** [almende/vis#1359]
The example html content item 7 contains a link that does not work on my Ipad (it works on desktop with a mouse), click command does not result in page request (see: http://visjs.org/examples/timeline/items/htmlContents.html).

Anyone an idea how I can solve this?

**Comment by josdejong** [comment]
Thanks for bringing this up, it's a known issue without a known solution. It has to do with the touch events (handled by hammer.js) being handled before the default browser behavior of opening the link.

A possible work around could be not rely on the default behavior of the browser, and instead, create a JavaScript click event to the links instead, like `onclick="window.open('http://visjs.org')"`.

I'm just thinking: maybe for a real solution we can mimic the default browser behavior: on a tap, check whether the tap occurred on an anchor element, and if so, open the url of the anchor using JavaScript.

**Comment by schotman** [comment]
Yes, this is about what I did. I changed the anchor in the item and added a onSelect function. Since in my case there is only one link per item, this was pretty easy. The onSelect function simple does a window.location="url belonging to the item".

**Comment by josdejong** [comment]
That's a nice workaround, but using `select` this is not applicable for all cases.

**Comment by schotman** [comment]
Yes, I understand, It works in my case because redirecting is the only thing that needs to be done.

**Comment by josdejong** [comment]
I'm not sure when what fixed this, but the bug seems not to appear anymore. I will close this issue.

**Comment by feus4177** [comment]
I'm using v4.21.0 and this seems to be a problem still. I can't rely on the `select` event workaround because I am already using it for something else. Any other ideas?

**Comment by xPapay** [comment]
I had same issue on touch devices. Click event never got fired event thought on desktop it worked as it was supposed to. I couldn't use select event neither because I've already use it for something else.
I've solved it by attaching `touchstart` event on my item (anchor tag). Since there it's easy, just read href attribute and set window location. It can be also used when element contains multiple buttons and attach each button different listener.
Hope it helps somebody.
```
element.addEventListener('touchstart', e => {
  const href = e.targetTouches[0].target.href;
  if (href) {
    window.location = href;
  }
});
```
